assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -657,7 +667,9 @@ const ExternalActivity: React.FC<PartComponentProps<CapiIframeModel>> = (props)
     // taken from simcapi.js TODO move somewhere, use from settings
     simLife.handshake.config = {
       context: context,
-      lessonId,
+      lessonId: randomizeLessonId
+        ? `${lessonId}_${Math.floor(Math.random() * 100)}_${Math.floor(Math.random() * 100)}`
```

Comment:
using a random number seems like it has the potential for creating a needle in a haystack bug. I presume that while the chance is very small, it still exists that the random number will choose the same number as before. so you will have a really small chance of this thing failing and it will be very hard to know why. so I suggest what about just increasing it? or maybe putting attempt number or guid on the end? so like `${lessonId}_${previousIndex + 1}` or something?

Reply:
also in general I don't really like the "IFRAME" so much because all of those other variables are something that affects the actual iframe so the purpose is clear... this is something different so it could potentially be a different key since it really just has to be something that the sim itself wouldn't set naturally. so maybe even `LESSON_ID_UNIQUE` ?

Reply:
I have updated the code base to use the` 'guid'` as well as changed the variable to `LESSON_ID_UNIQUE`